apps/passport-client/components/screens/ProveScreen/GenericProveScreen.tsx

```diff
@@ -25,11 +26,13 @@ export function GenericProveScreen({ req }: { req: PCDGetRequest }) {
   const [state, dispatch] = useContext(DispatchContext);
   const [args, setArgs] = useState(JSON.parse(JSON.stringify(req.args)));
   const [error, setError] = useState<Error | undefined>();
+  const [proving, setProving] = useState(false);
 
   const pcdPackage = state.pcds.getPackage(req.pcdType);
 
   const onProveClick = useCallback(async () => {
     try {
+      setProving(true);
```

Comment:
One thing I recommend doing is adding an `await sleep(200)` (you'll have to copy the implementation of `sleep` from the `passport-server` project) between line 45 and 46 of this file, so that the UI has a chance to update to the 'loading' state before the potentially blocking proving operation kicks off.

Plz make that change in the rest of the proving screens too.